### PR TITLE
refract wai-logger

### PR DESCRIPTION
**Pull request by winterland1989**
This is the continue of #84 , after this get merged, @kazu-yamamoto please release a new version fast-logger and wai-logger with version bumped.

### CODE REVIEW

File: fast-logger/System/Log/FastLogger/Date.hs [hunk 1]
```diff
@@ -55,7 +57,14 @@ formatDate fmt = formatUnixTime fmt . fromEpochTime
...
+-- auto updating formatted time, this cache update every 1 second.
+-- more detail in "Control.AutoUpdate"
+newTimeCache :: TimeFormat -> IO (IO FormattedTime)
+newTimeCache fmt = mkAutoUpdate defaultUpdateSettings{
         updateAction = getTime >>= formatDate fmt
     }
+
+-- | A simple time cache using format @"%d/%b/%Y:%T %z"@
+simpleTimeCache :: IO FormattedTime
+simpleTimeCache = newTimeCache "%d/%b/%Y:%T %z" >>= id
```
Comment (winterland1989) [inline fast-logger/System/Log/FastLogger/Date.hs]: This seemed not right here, every time i extra cache from `simpleTimeCache`, do i run `newTimeCache` again? or should i use `unsafePerformIO` tricks here?

File: fast-logger/System/Log/FastLogger/Date.hs [hunk 2]
```diff
@@ -55,7 +58,15 @@ formatDate fmt = formatUnixTime fmt . fromEpochTime
...
+-- more detail in "Control.AutoUpdate"
+newTimeCache :: TimeFormat -> IO (IO FormattedTime)
+newTimeCache fmt = mkAutoUpdate defaultUpdateSettings{
         updateAction = getTime >>= formatDate fmt
     }
+
+-- | A simple time cache using format @"%d/%b/%Y:%T %z"@
+simpleTimeCache :: IO FormattedTime
+simpleTimeCache = unsafePerformIO $ newTimeCache "%d/%b/%Y:%T %z"
+{-# NOINLINE simpleTimeCache #-}
```
Comment (winterland1989) [inline fast-logger/System/Log/FastLogger/Date.hs]: Now i understand what IO is : )
If you don't like this approach, we can leave cache initialization  to user.

Comment (kazu-yamamoto) [inline fast-logger/System/Log/FastLogger/Date.hs]: You use `NOINLINE` correctly. You know what `unsafePerformIO` is. :-)

Comment (kazu-yamamoto) [inline fast-logger/System/Log/FastLogger/Date.hs]: I guess that this global cacher can be easily implemented by users. So, I cannot find good reasons to define this global cacher in this package.

Comment (winterland1989) [inline fast-logger/System/Log/FastLogger/Date.hs]: Ok, i will remove it and add `simpleTimeFormat` back.

File: fast-logger/System/Log/FastLogger/Date.hs [hunk 3]
```diff
@@ -67,6 +67,9 @@ newTimeCache fmt = mkAutoUpdate defaultUpdateSettings{
...
 -- | A simple time cache using format @"%d/%b/%Y:%T %z"@
-simpleTimeCache :: IO FormattedTime
-simpleTimeCache = unsafePerformIO $ newTimeCache "%d/%b/%Y:%T %z"
-{-# NOINLINE simpleTimeCache #-}
+simpleTimeFormat :: TimeFormat
+simpleTimeFormat = "%d/%b/%Y:%T %z"
+
+-- | A simple time cache using format @"%d-%b-%Y %T"@
+simpleTimeFormat' :: TimeFormat
+simpleTimeFormat' = "%d-%b-%Y %T"
```
Comment (winterland1989) [inline fast-logger/System/Log/FastLogger/Date.hs]: I took the liberty to add a format i often use here ; )

**Comment by winterland1989** [comment]
After this get merged, i will continue make a pr to monad-logger with theres helpers:

```haskell
runFastLoggerLoggingT :: MonadIO m => FastLogger -> LoggingT m a -> m a
runTimedFastLoggerLoggingT :: MonadIO m => TimedFastLogger -> LoggingT m a -> m a
```
Should we release them together?

**Comment by kazu-yamamoto** [comment]
LGTM. Merged.

I will review this again tomorrow before releasing.

**Comment by winterland1989** [comment]
Thanks!

**Pull request merged**